spec/calculator_spec.rb

```diff
@@ -1 +1,259 @@
+	      		expect(@test_calc.divide(77, 11)).to eq(7)
+	    	end
+
+	    	it 'returns a positive integer when there is no remainder with negative, positive integer arguments' do
+	      		expect(@test_calc.divide(-12, 3)).to eq(-4)
+	    	end
+
+	    	it 'returns a positive integer when there is no remainder with positive, negative integer arguments' do
+	      		expect(@test_calc.divide(10, -2)).to eq(-5)
+	    	end
+
+	    	it 'returns a positive integer when there is no remainder with negative integer arguments' do
+	      		expect(@test_calc.divide(-30, -6)).to eq(5)
+	    	end
+
+	    	it 'returns an integer when there is no remainder with 2 floats' do
+	    		first_arg = -4.42
+	    		second_arg = 2.21
+	      		expect(@test_calc.divide(first_arg, second_arg)).to be_within(0.01).of(-2)
+	    	end
+
+	    	it 'returns a float when there is remainder with 2 floats' do
+	    		first_arg = -4.41
+	    		second_arg = 2.21
+	      		expect(@test_calc.divide(first_arg, second_arg)).to be_within(0.01).of(-2.0)
+	    	end
+
+	    	it "can divide very big numbers" do
+				first_arg = 8972348972348
+	    		second_arg = 90544564353
+	      		expect(@test_calc.divide(first_arg, second_arg)).to be_within(0.01).of(8972348972348.0/90544564353.0)
+	    	end
+	  	end
+
+	  	describe '#pow' do
+			
+			it "can calculate a positive integer power of a number" do
+				first_arg = 11
+	    		second_arg = 2
+	      		expect(@test_calc.pow(first_arg, second_arg)).to eq(121)
+			end
+			
+			it "can calculate large values" do
+				first_arg = 5
+	    		second_arg = 7
+	      		expect(@test_calc.pow(first_arg, second_arg)).to eq(5**7)
+			end
+
+			it "can calculate negative powers" do
+				first_arg = 5
+	    		second_arg = -7
+	      		expect(@test_calc.pow(first_arg, second_arg)).to eq(5**-7)
+			end
+
+			it "can calculate decimal powers" do
+				first_arg = 6
+	    		second_arg = 1.7
+	      		expect(@test_calc.pow(first_arg, second_arg)).to eq(6**1.7)
+			end
+	  	end
+
+	  	describe '#sqrt' do
+		
+			it "can calculate sqr root of a positive integer" do
+				first_arg = 225
+	      		expect(@test_calc.sqrt(first_arg)).to eq(15)
+			end
+
+			it "can calculate sqr root of a positive integer in 2 decimal digits" do
+				first_arg = 224
+	      		expect(@test_calc.sqrt(first_arg)).to be_within(0.1).of(15)
+			end
+			
+			it "throws error if argument is negative" do
+				first_arg = -5
+	      		expect{@test_calc.sqrt(first_arg)}.to raise_error(ArgumentError)
+			end
+	  	end
+
+	  	describe '#memory' do
+	  		it "setter, overrides previous object in memory" do
+	  			before_memory_instance = @test_calc.memory
+	  			@test_calc.memory = 25
+	  			after_memory_instance = @test_calc.memory
+	  			expect(before_memory_instance).not_to eq(after_memory_instance)
+	  		end
+
+	  		it "getter, returns the memory object" do
+	  			@test_calc.memory = 13
+				memory_instance = @test_calc.memory
```

Comment:
this line is redundant, you can directly use `@test_calc.memory` in the next line.